src/dos/drive_virtual.cpp

```diff
@@ -145,12 +153,11 @@ void VFILE_Register(const char *name,
 		if (onpos == 0)
 			return;
 	}
-	auto cur_file = first_file;
-	while (cur_file) {
-		if (onpos == cur_file->onpos && iequals(name, cur_file->name)) {
-			return;
-		}
-		cur_file = cur_file->next;
+	auto found = find_file_by_predicate(first_file, [onpos, name](vfile_block_t cur_file) {
+		return onpos == cur_file->onpos && iequals(name, cur_file->name);
+	});
```

Comment:
I'm finding the two embedded lamda prototypes combined with `find_file_by_predicate` visually harder to read than before:
![2022-12-14_09-33](https://user-images.githubusercontent.com/1557255/207666709-ddb096e6-4b45-435c-a9e9-8f6b7eee279f.png)

Can these two find_file_by_predicate + lamdas be wrapped into a a single helper function? `bool vfile_name_and_pos_exists(name, pos)`?

It could then be used like:

line 120: 
```c++
if (vfile_name_and_pos_exists(sfn, onpos)) {
    return;
}
```

line 156:
```c++
if (vfile_name_and_pos_exists(name, onpos)) {
    return;
}
```

Reply:
good idea